From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Xiaotian Wu <wuxiaotian@loongson.cn>
Date: Mon, 1 Sep 2025 17:00:04 +0800
Subject: [PATCH] Do not generate SIMD on LoongArch architecture

---
 efi/meson.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/efi/meson.build b/efi/meson.build
index 36dfa5d..ae40d77 100644
--- a/efi/meson.build
+++ b/efi/meson.build
@@ -197,6 +197,9 @@ elif host_cpu == 'x86'
                    '-mno-mmx',
                    '-mno-red-zone',
                    '-m32']
+elif host_cpu == 'loongarch64'
+  compile_args += ['-mno-lsx',
+                   '-mno-lasx']
 # no special cases for aarch64 or arm
 endif
 
-- 
2.47.3

